packages/tooling/fast-tooling-react/src/form/custom-controls/control.css.md

```diff
@@ -0,0 +1,140 @@
+
+## Variables
+
+There should be a list of internal variables that are assigned to each CSS property. The structure should look like this:
+
+```typescript
+[
+    {
+        "border": { // The property name should be pascalCase
+            name: "border", // The name should be spinal-case
+            shorthand: true, // denotes that "border" CSS property is a shorthand
+            default: "none",
+            globalValues: [
+                {
+                    value: "inherit"
+                },
+                {
+                    value: "initial"
+                },
+                {
+                    value: "unset"
+                }
+            ],
+            values: [ // these must be in order
+                {
+                    mapsToCssProperty: "border-width",
+                    type: "width",
+                    default: "1px"
+                },
+                {
+                    mapsToCssProperty: "border-style",
+                    type: "border-style",
+                    default: "solid"
+                },
+                {
+                    mapsToCssProperty: "border-color",
+                    type: "color",
+                    default: "black"
+                }
+            ]
+        }
+    },
+    // Complete list of all potential CSS properties
+]
+```
+
+From this structure we can construct via the UI generated from the below methods: `border: 1px solid red`.
+
+## Methods
+
+### Render methods
+
+Render methods should be created for the following inputs:
+- Text field
+    - Used as a default value if the property name is not recognized
+- Autocomplete dropdown
+    - Used by the property name selector, once this has been chosen, the value should reflect different inputs
+    - Used by color (full list of available colors can be autocompleted)
+    - Used by border-style
+    - Used by anything which could use a CSS function such as var()
+- Number field + Autocomplete dropdown
+    - Used by width (the select will be populated by `px`, `em`, etc.)
+
+Further rules will need to be defined based on [MDN CSS syntax](https://github.com/mdn/data/blob/master/css/syntaxes.json).
+
+### Utility methods
+
+Data conversion utilities should exist to transform strings to objects and objects to strings. Objects will be stored as part of the components state, while strings will be returned during update.
+Some utilities that may be required:
+- `convertObjectToCSS` - convert an object + selectors to a CSS string, if no selectors are available convert it to the inline equivalent.
+- `convertCSSToObject` - convert a string to an object + selectors (if available)
+
+## Overrides
+
+To make this flexible, the abstract class should allow a way to override the default creation of any property. The implementation of this in the CSS inline or CSS internal controls may look like:
+
+```tsx
+<CSSInline
```

Comment:
I wonder if we could start introducing web components for any new controls...

Reply:
I thought about that, but I think introducing a wrapper, among other things, plus the additional testing, is going to be more of a headache than just having two sets and eventually deprecating the React version. I'm keeping the React components as web component-like as possible as an intermediary step.

Reply:
Open for discussion on that though, I'm just not sure the payoff is worth it.

Reply:
Deferring to your judgement on when the right time is for that.